duckduckgo/macos-browser
# Use Settings instead of Preferences on Bitwarden dialog when targeting macOS 13 and +

# Task/Issue URL 
 
https://app.asana.com/0/1177771139624306/1204034291081092/f

## Description
The `Preferences` app menu item text changed in Ventura (13.0) to `Settings`. So when targeting Ventura and (probably) upcoming releases, we need to show `Settings` instead of `Preferences` when pointing to an application settings menu item.

When opening the pull request (and because this happened in the [past](https://github.com/duckduckgo/macos-browser/pull/925)), I thought of another solution to encapsulate the logic of having different strings for older versions than Ventura:

```swift
static let bitwardenSettingsLocalized = localizeForVersionsOlderThanVentura(
        key: "select.bitwarden.settings",
        current: "Select Bitwarden → Preferences from the Mac menu bar.",
        old: "Select Bitwarden → Settings from the Mac menu bar.",
        comment: "Setup of the integration with Bitwarden app"
    )
    
private static func localizeForVersionsOlderThanVentura(
      key: String,
      current: String,
      old: String,
      comment: String
  ) -> String {
      if #available(macOS 13.0, *) {
          return NSLocalizedString(key, value: current, comment: comment)
      } else {
          return NSLocalizedString(key, value: old, comment: comment)
      }
  }
```

That way, we do not have to check for macOS versions in the view code. We can just do the following:
```swift
Text(UserText.bitwardenSettingsLocalized)
```
I went with the current solution because it is something that only happened twice. If we start seeing this issue more often, revisiting this approach and getting something more generic and encapsulated would be nice.

## Steps to test this PR
1. Install the Bitwarden macOS application
2. Open the DDG macOS browser and tap the three-dot button (more button)
3. Go to Autofill and select Bitwarden below the Password Manager options
4. A dialog will pop up to set up Bitwarden
5. You should see the following text in step 2 of the dialog: 'Bitwarden -> Preferences' when testing below Ventura (13.0) and 'Bitwarden -> Settings' when testing Ventura and above.

## Screenshots

| Ventura (macOS 13) | Monterrey (macOS 12) |
| --- | --- |
| ![Ventura - Bitwarden](https://user-images.githubusercontent.com/7924732/221901220-cfbd1f97-5ee1-496e-8688-35bf4e093a54.png)  | ![Monterrey - Bitwarden](https://user-images.githubusercontent.com/7924732/221900105-447265f2-5307-4f2a-833c-f7ea483ee75a.png) |
